daemon/create.go

```diff
@@ -78,6 +78,21 @@ func (daemon *Daemon) containerCreate(params types.ContainerCreateConfig, manage
 	return containertypes.ContainerCreateCreatedBody{ID: container.ID, Warnings: warnings}, nil
 }
 
+func setContainerIDEnv(container *container.Container, cidenv string) error {
+	if cidenv == "" {
+		return nil
+	}
+	// check if we already have an env var like that
+	for _, env := range container.Config.Env {
+		if strings.HasPrefix(env, cidenv) {
+			return fmt.Errorf("Error environment variable %s already defined", cidenv)
+		}
+	}
+	container.Config.Env = append(container.Config.Env, fmt.Sprintf("%s=%s", cidenv, container.ID))
+	logrus.WithField("module", "cidenv").Debugf("setting %s=%s", cidenv, container.ID)
```

Comment:
Not sure if `module` is the best field name here. Because cidenv is not really a module, but just a feature extension. Do we use similar logging in other places we extend the engine functionality?

Reply:
agreed. I'm not even sure we need to log this at all. I guess I will just remove the field and make the message something like:
```adding %s=%s to container environment```

Reply:
In fact, it may be worth logging of our engine extensions.
Maybe just changing the field:
```
logrus.WithField("tag", "balena-extension")
```
It would help wot quickly identify logs produced by non-standard moby code.

Reply:
Sounds reasonable.